From 196b79b04126061adadb6f55dd9467dfe8940289 Mon Sep 17 00:00:00 2001
From: derek fang <derek.fang@realtek.com>
Date: Fri, 10 Apr 2026 16:10:48 +0800
Subject: [PATCH] sof-soundwire: add companion amp config with rt721 and rt1320

Support the machines with companion amps with rt721 and rt1320.

Closes: https://github.com/alsa-project/alsa-ucm-conf/pull/744/
Signed-off-by: derek fang <derek.fang@realtek.com>
Signed-off-by: Jaroslav Kysela <perex@perex.cz>
Origin: upstream, https://github.com/alsa-project/alsa-ucm-conf/commit/196b79b04126061adadb6f55dd9467dfe8940289
Bug-Ubuntu: https://bugs.launchpad.net/bugs/2154418
---
 ucm2/sof-soundwire/rt721+rt1320.conf | 51 ++++++++++++++++++++++++++++
 1 file changed, 51 insertions(+)
 create mode 100644 ucm2/sof-soundwire/rt721+rt1320.conf

diff --git a/ucm2/sof-soundwire/rt721+rt1320.conf b/ucm2/sof-soundwire/rt721+rt1320.conf
new file mode 100644
index 0000000..4fcad9c
--- /dev/null
+++ b/ucm2/sof-soundwire/rt721+rt1320.conf
@@ -0,0 +1,51 @@
+# Use case Configuration for sof-soundwire card
+
+#
+# Arguments:
+#   Amp - amplifier number (1,2 etc.)
+#   Sel - channel selection (L,R R,L L,L R,R L,L+R R,L+R L+R,L L+R,R L+R,L+R)
+#
+DefineMacro.rt1320spk {
+	EnableSequence [
+		cset "name='rt1320-${var:__Amp} RX Channel Select' ${var:__Sel}"
+		cset "name='rt1320-${var:__Amp} OT23 L Switch' 1"
+		cset "name='rt1320-${var:__Amp} OT23 R Switch' 1"
+	]
+	DisableSequence [
+		cset "name='rt1320-${var:__Amp} OT23 L Switch' 0"
+		cset "name='rt1320-${var:__Amp} OT23 R Switch' 0"
+	]
+}
+
+DefineMacro.rt1320AmpNum.If.0 {
+	Condition {
+		Type ControlExists
+		Control "name='rt1320-2 OT23 L Switch'"
+	}
+	True {
+		Macro.num1.rt1320spk { Amp 1 Sel "L,L" }
+		Macro.num2.rt1320spk { Amp 2 Sel "R,R" }
+	}
+	False {
+		Macro.num1.rt1320spk { Amp 1 Sel "L,R" }
+	}
+}
+
+SectionDevice."Speaker" {
+	Comment	"Speaker"
+
+	Macro.num1.rt1320AmpNum { }
+
+	EnableSequence [
+		cset "name='Speaker Switch' on"
+	]
+
+	DisableSequence [
+		cset "name='Speaker Switch' off"
+	]
+
+	Value {
+	      PlaybackPriority 100
+	      PlaybackPCM "hw:${CardId},2"
+	}
+}
-- 
2.43.0

